### DISCUSSION TITLE
newbie questions about .dwg (pages and external references)

### DISCUSSION THREAD

**Discussion by abubelinha** [LibreDWG/libredwg#585] [Q&A] [answered]
I am totally agnostic about AutoCAD or any CAD programs, but I got a .dwg file made in AutoCAD 2022 which I need to open either in a GIS (ideally Quantum GIS) or in some other way so that I can extract data.

Using some of the [Autodesk-available viewers](https://www.autodesk.com/viewers), I realized that: 
1. The .dwg file shows different views of a house, separated in several "pages" (sorry, don't know their technical name inside AutoCAD:  Autodesk _online viewer_ calls them "sheets" but their Windows _DWG TrueView_ viewer calls them "pages").
2. One of those "pages" shows external raster images (aerial photo background) which map owner did not send to me (not important for the map info). 

Should any of those cause troubles to LibreDWG? I wonder if there is any way to tell LibreDWG:
(1) Specify which one of the pages you want to convert to i.e. geojson or svg  
(2) Get rid of external file references (remove them so output files have no other dependencies).

Thanks

**Comment by rurban** [comment, marked as answer]
LibreDWG only has rudimentary viewer support yet. Its SVG or PS output misses a lot of conversions and entities. 
And you cannot specify which LAYOUT or PAPERSPACE view to output. (I think that's what they call sheets or pages).

Would be trivial to implement, if someone wants though. Just not our priority yet.

**Comment by abubelinha** [comment]
Thanks for reply.
So, when a .dwg contains different LAYOUT/PAPERSPACE ... how does LibreDWG handle them?
i.e., if I convert .dwg to geojson or json using LibreDWG, what's the expected output?
Would LibreDWG output each layout as a different json file?

Sorry if this is a silly question. I just received my file from someone else but I don't know if it is an uncommon situation that .dwg contains several layouts

**Reply by rurban** [reply]
LibreDWG's decoder does nothing much. It simply represents the DWG internal structure. You need to understand it first, and it's not high-level.

**Reply by abubelinha** [reply]
Thanks but I don't get your point.

As I don't understand DWG internals, that's exactly why I want to convert to another format.
My file contains several maps, so I think geojson (for example) would be OK. 
I mean something like this:     `     dwgread -o myoutput.geojson myfile.dwg`
Then I could import each geojson map to a GIS, or at least I could view contents and extract the info I need.

So @rurban ... are you suggesting me not to do that?
Do you mean I should find better options than LibreDWG for exporting a multi-layout DWG to geojson (or any other format)?
